app/scripts/modules/docker/src/pipeline/trigger/DockerTriggerTemplate.tsx

```diff
@@ -70,21 +70,29 @@ export class DockerTriggerTemplate extends React.Component<
     this.setState({ lookupType: newType });
   };
 
-  private updateArtifact(command: IPipelineCommand, tag: string) {
-    this.props.updateCommand('extraFields.tag', tag);
+  private updateArtifact(command: IPipelineCommand, tagOrDigest: string) {
```

Comment:
One thing I noticed while testing this is that we don't call `updateArtifact` when we toggle between `tag` and `digest` lookup types. This seems problematic because it's possible to get into a state where `extraFields.tag` points to your tag as expected, but `extraFields.artifacts[0]` still references a digest you had entered previously. It seems like what we actually want to do is replace this line from `lookupTypeChanged`:

`this.props.updateCommand('extraFields.tag', newType === 'tag' ? this.state.selectedTag : this.state.digest);`

with something like this:

`this.updateArtifact(this.props.command, newType === 'tag' ? this.state.selectedTag : this.state.digest)`

What do you think?

Reply:
Thank you for the comment.

Yes, you are right.
It's better to update the `lookupTypeChanged` to replace `this.props.updateCommand(...` with `this.updateArtifact(...`.

I'm on it 👍

Reply:
Fixed in https://github.com/spinnaker/deck/pull/7896/commits/1642c2a7afcb82cc49b9f1b5abefa649e5f35782

Reply:
Thanks for the fix; looks good to me!